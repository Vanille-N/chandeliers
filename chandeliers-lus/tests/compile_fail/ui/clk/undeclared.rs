//! `when` uses an undeclared variable as clock.
chandeliers_lus::decl! {
    node foo(b : bool; i : int when b2) returns ();
    let tel;
}

fn main() {}
