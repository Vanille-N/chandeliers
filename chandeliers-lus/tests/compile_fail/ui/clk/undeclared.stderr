error: Variable b2 is not available yet at this point of the typechecking.
 --> tests/compile_fail/ui/clk/undeclared.rs:3:37
  |
3 |     node foo(b : bool; i : int when b2) returns ();
  |                                     ^^
  |
  = note: During this incremental typechecking, you cannot access global
          variables and you may only use local variables that have been
          declared strictly beforehand, in the order of inputs then outputs
          then locals.
  = note: These are the variables that are already useable: b
